auth0/v3/test/authentication/test_base.py

```diff
@@ -50,10 +50,25 @@ def test_post(self, mock_post):
 
         data = ab.post('the-url', data={'a': 'b'}, headers={'c': 'd'})
 
-        mock_post.assert_called_with(url='the-url', data='{"a": "b"}',
+        mock_post.assert_called_with(url='the-url', json={'a': 'b'},
                 headers={'c': 'd', 'Content-Type': 'application/json'})
 
         self.assertEqual(data, {'x': 'y'})
+        
+    @mock.patch('requests.post')
+    def test_post_with_defaults(self, mock_post):
+        ab = AuthenticationBase('auth0.com', telemetry=False)
+
+        mock_post.return_value.status_code = 200
+        mock_post.return_value.text = '{"x": "y"}'
+
+        # Only required params are passed
+        data = ab.post('the-url')
```

Comment:
no `headers` value is passed, so default of None will kick in. This asserts the bug is no longer present